src/main/java/org/ibp/api/java/impl/middleware/study/StudyInstanceServiceImpl.java

```diff
@@ -117,66 +116,102 @@ public void deleteStudyInstances(final Integer studyId, final List<Integer> inst
 	}
 
 	@Override
-	public InstanceData addInstanceData(final Integer studyId, final Integer instanceId, final InstanceData instanceData) {
+	public ObservationData addInstanceObservation(final Integer studyId, final Integer instanceId, final ObservationData observationData) {
 		this.studyValidator.validate(studyId, true);
 		this.instanceValidator.validateStudyInstance(studyId, Collections.singleton(instanceId));
 
-		final Integer datasetId = this.getEnvironmentDatasetId(studyId);
```

Comment:
The common validations for add/update Instance observation can be consolidated to a method:
```
this.studyValidator.validate(studyId, true);
this.instanceValidator.validateStudyInstance(studyId, Collections.singleton(instanceId));

final Integer datasetId = this.studyService.getEnvironmentDatasetId(studyId);
final Integer variableId = observationData.getVariableId();
this.datasetValidator.validateExistingDatasetVariables(studyId, datasetId, Collections.singletonList(
			variableId));
this.observationValidator.validateVariableValue(variableId, observationData.getValue());
this.datasetValidator.validateVariableBelongsToVariableType(datasetId, variableId, VariableType.STUDY_CONDITION.getId());
```